cloudsql-mysql-plugin/src/main/java/io/cdap/plugin/cloudsql/mysql/CloudsqlMysqlSink.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.cloudsql.mysql;
+
+import com.google.common.collect.ImmutableMap;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.etl.api.batch.BatchSink;
+import io.cdap.plugin.db.CommonSchemaReader;
+import io.cdap.plugin.db.SchemaReader;
+import io.cdap.plugin.db.batch.sink.AbstractDBSink;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/** Sink support for a CloudSQL MySQL database. */
+@Plugin(type = BatchSink.PLUGIN_TYPE)
+@Name(CloudsqlMysqlConstants.PLUGIN_NAME)
+@Description(
+    "Writes records to a CloudSQL MySQL table. Each record will be written in a row in the table.")
+public class CloudsqlMysqlSink extends AbstractDBSink {
+
+  private final CloudsqlMysqlSinkConfig cloudsqlMysqlSinkConfig;
+
+  public CloudsqlMysqlSink(CloudsqlMysqlSinkConfig cloudsqlMysqlSinkConfig) {
+    super(cloudsqlMysqlSinkConfig);
+    this.cloudsqlMysqlSinkConfig = cloudsqlMysqlSinkConfig;
+  }
+  
+  @Override
+  protected SchemaReader getSchemaReader() {
+    return new CommonSchemaReader();
+  }
+  
+  /** CloudSQL MySQL sink configuration. */
+  public static class CloudsqlMysqlSinkConfig extends AbstractDBSink.DBSinkConfig {
```

Comment:
I think we should create a base class called CloudSQLMySQLConfig, and have the source, sink and action configs extend from it. Otherwise there seems to be a lot of repetition.

Reply:
Makes sense. I'll create a base config class.